dask_cuda/tests/test_utils.py

```diff
@@ -245,3 +245,30 @@ def test_parse_device_memory_limit():
     assert parse_device_memory_limit(0.8) == int(total * 0.8)
     assert parse_device_memory_limit(1000000000) == 1000000000
     assert parse_device_memory_limit("1GB") == 1000000000
+
+
+def test_parse_visible_mig_devices():
+    pynvml = pytest.importorskip("pynvml")
+    pynvml.nvmlInit()
+    for index in range(get_gpu_count()):
+        handle = pynvml.nvmlDeviceGetHandleByIndex(index)
+        try:
+            mode = pynvml.nvmlDeviceGetMigMode(handle)[0]
+        except pynvml.NVMLError:
+            # if not a MIG device, i.e. a normal GPU, skip
+            continue
+        if mode:
+            # Just checks to see if there are any MIG enabled GPUS
+            # If there is one, check if the number of mig enabled
+            # instances is less than 7
+            count = pynvml.nvmlDeviceGetMaxMigDeviceCount(handle)
+            miguuids = []
+            for i in range(count):
+                try:
+                    mighandle = pynvml.nvmlDeviceGetMigDeviceHandleByIndex(
+                        device=handle, index=i
+                    )
+                    miguuids.append(mighandle)
+                except pynvml.NVMLError:
+                    pass
+            assert len(miguuids) <= 7
```

Comment:
Why this magical number `7`? What if there are `8` or more?

Reply:
A more accurate assertion would probably be `assert len(miguuids) == count`.

Reply:
I added 7 because you can only partition a single MIG enabled GPU into atmost 7 independent MIG instances. But there may be less in reality. `nvmlDeviceGetMaxMigDeviceCount` gives us the maximum number of MIG devices/instances that can exist under a given parent NVML device, not the actual number currently present.

Reply:
This is only true today, you never know if this is still going to be the case in a future GPU, so we have to consider that instead of relying on what holds true today.